src/ndpp_init.F90

```diff
@@ -0,0 +1,622 @@
+       i = index(ndpp_lib, "/", BACK=.true.)
+       directory = ndpp_lib(1:i)
+    end if
+
+    ! determine whether binary, ascii, or hdf5
+    temp_str = ''
+    if (check_for_node(doc, "filetype")) &
+      call get_node_value(doc, "filetype", temp_str)
+    if (trim(temp_str) == 'ascii') then
+       filetype = ASCII
+    elseif (trim(temp_str) == 'binary') then
+       filetype = BINARY
+    elseif (trim(temp_str) == 'hdf5') then
+       filetype = H5
+    else
+       call fatal_error("Unknown filetype in " // trim(ndpp_lib) // &
+            ": " // trim(temp_str))
+    end if
+
+    ! Test metadata to ensure this library matches the problem definition
+    ! First test to see if we have a legendre scattering type and not tabular
+    ! (This will be removed at an undefined date when OpenMC supports
+    ! tabular scattering distributions as opposed to just Legendres.
+    ! Note that the authors have no immediate plans for this).
+    if (check_for_node(doc, "scatt_type")) &
+      call get_node_value(doc, "scatt_type", scatt_type)
+    if (scatt_type /= SCATT_TYPE_LEGENDRE) then
+      call fatal_error("Invalid Scattering Type represented in NDPP " // & 
+           "data. Rerun NDPP with the Legendre scattering type set.")
+    end if
+
+    ! Test to ensure the scattering order is less than the maximum
+    order = 0
+    if (check_for_node(doc, "scatt_order")) &
+      call get_node_value(doc, "scatt_order", order)
+    if (order > MAX_ANG_ORDER) then
+      call fatal_error("Invalid scattering order of " // &
+           trim(to_str(order)) // " requested.")
+    end if
+
+    ! Get the energy bin structure
+    if (check_for_node(doc, "energy_bins")) then
+      ndpp_groups = get_arraysize_double(doc, "energy_bins")
+      allocate(ndpp_energy_bins(ndpp_groups))
+      ndpp_groups = ndpp_groups - 1
+      call get_node_array(doc, "energy_bins", ndpp_energy_bins)
+    end if
+
+    ! Get if nuscatter is present
+    temp_str = ""
+    if (check_for_node(doc, "nuscatter")) then
+      call get_node_value(doc, "nuscatter", temp_str)
+      if (trim(temp_str) == 'true') then
+        nuscatter = .true.
+      else
+        nuscatter = .false.
+      end if
+    else
+      nuscatter = .false.
+    end if
+
+    ! Get if chi is present
+    temp_str = ""
+    if (check_for_node(doc, "chi_present")) then
+      call get_node_value(doc, "chi_present", temp_str)
+      if (trim(temp_str) == 'true') then
+        chi_present = .true.
+      else
+        chi_present = .false.
+      end if
+    else
+      chi_present = .false.
+    end if
+
+    ! Check the tallies to ensure that the energy group structure, scattering
+    ! order requested are valid (i.e., groups match, orders are less than in
+    ! the library)
+    TALLY_LOOP: do i = 1, n_tallies
+      t => tallies(i)
+      j = 0
+      SCORE_LOOP: do k = 1, t % n_user_score_bins
+        j = j + 1
+        select case (t % score_bins(j))
+          case (SCORE_SCATTER_PN)
```

Comment:
Is there any particular reason you specify a case for `SCORE_SCATTER_PN`, but none of the other non-NDPP scores?  Honestly I don't think the extra code for the non-NDPP cases is worth the computation time that it saves.  I think you should just remove this case.